generator/src/main/java/com/scottlogic/deg/generator/decisiontree/MostProlificConstraintOptimiser.java

```diff
@@ -211,9 +215,14 @@ DecisionAnalysisResult performAnalysis() {
             return result;
         }
 
-        private void markOptionForFactorisation(AtomicConstraint factorisingConstraint, ConstraintNode node, List<ConstraintNode> options, Set<AtomicConstraint> constraints){
-            ConstraintNode newOption = node.cloneWithoutAtomicConstraint(factorisingConstraint);
-            if (!newOption.getAtomicConstraints().isEmpty()){
+
+
+        private void markOptionForFactorisation(AtomicConstraint factorisingConstraint,
+                                                ConstraintNode node,
+                                                List<ConstraintNode> options,
+                                                Set<AtomicConstraint> constraints) {
+            ConstraintNode newOption = node.builder().removeAtomicConstraint(factorisingConstraint).build();
```

Comment:
It might be worth formatting this method something like this, to be more consistent with how we are doing it elsewhere:
```
        private void markOptionForFactorisation(
            AtomicConstraint factorisingConstraint,
            ConstraintNode node,
            List<ConstraintNode> options,
            Set<AtomicConstraint> constraints) {
```